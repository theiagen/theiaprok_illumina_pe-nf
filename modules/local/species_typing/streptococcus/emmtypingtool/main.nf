process EMMTYPINGTOOL {
    tag "$meta.id"
    label "process_low"

    container "us-docker.pkg.dev/general-theiagen/staphb/emmtypingtool:0.0.1"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*.emmtypingtool.xml"), emit: emmtypingtool_results
    tuple val(meta), path("EMM_TYPE"), emit: emmtypingtool_emm_type
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    if (reads.size() != 2) {
        error "EMMTYPINGTOOL requires exactly two input reads (paired-end). Found: ${reads.size()}"
    }
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.2.0' // No option to get version from tool
    
    
    """
    emm_typing.py \\
      -m /db \\
      -1 ${reads[0]} \\
      -2 ${reads[1]} \\
      -o output_dir \\
        ${args}

    grep "Final_EMM_type" output_dir/*.results.xml | sed -n 's/.*value="\\([^"]*\\)".*/\\1/p' | tee EMM_TYPE
    mv output_dir/*.results.xml ${prefix}.emmtypingtool.xml

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        emmtypingtool: ${VERSION}
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.2.0' // No option to get version from tool
    """
    touch ${prefix}.emmtypingtool.xml
    touch EMM_TYPE

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        emmtypingtool: ${VERSION}
    END_VERSIONS
    """
}