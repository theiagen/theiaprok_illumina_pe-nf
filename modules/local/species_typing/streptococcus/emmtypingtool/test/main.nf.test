nextflow_process {
    name "Test Process EMMTYPINGTOOL"
    script "../main.nf"
    process "EMMTYPINGTOOL"

    test("EMMTYPINGTOOL - reads paired-end") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    [
                        file("https://github.com/sanger-pathogens/seroba/raw/refs/heads/master/seroba/tests/data/06B_1.fq.gz", checkIfExists: true),
                        file("https://github.com/sanger-pathogens/seroba/raw/refs/heads/master/seroba/tests/data/06B_2.fq.gz", checkIfExists: true)
                    ]
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.emmtypingtool_results,
                    process.out.emmtypingtool_emm_type,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("EMMTYPINGTOOL - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    [
                        file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists: true),
                        file(params.pipelines_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists: true)
                    ]
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}