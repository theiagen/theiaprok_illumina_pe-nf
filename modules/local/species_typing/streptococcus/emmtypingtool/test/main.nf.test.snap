{
    "EMMTYPINGTOOL - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.emmtypingtool.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EMM_TYPE:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,558a35d46f06435747f1a3184c6aacc7"
                ],
                "emmtypingtool_emm_type": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "EMM_TYPE:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "emmtypingtool_results": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.emmtypingtool.xml:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,558a35d46f06435747f1a3184c6aacc7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-25T19:20:16.151069766"
    },
    "EMMTYPINGTOOL - reads paired-end": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.emmtypingtool.xml:md5,a55a59074358aa28dcf17337083dee77"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "EMM_TYPE:md5,1458e56a166c4bca1486d71611e8137f"
                ]
            ],
            [
                "versions.yml:md5,558a35d46f06435747f1a3184c6aacc7"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-06-25T19:20:07.320837958"
    }
}